src/components/BarChart.js

```diff
@@ -211,11 +211,26 @@ const BarChart = React.createClass({
       return d.value;
     });
     const gap = 0.3;
-    const y0 = Math.max(Math.abs(d3.min(data)), Math.abs(d3.max(data)));
-
+    const hasNegative = !!data.filter(value => value < 0).length;
+    const hasPositive = !!data.filter(value => value > 0).length;
+    let domain;
+    let range;
+
+    if (hasNegative && !hasPositive) {
+      domain = [d3.min(data), 0];
+      range = [-height, 0];
+    } else if (!hasNegative && hasPositive) {
+      domain = [0, d3.max(data)];
+      range = [0, height];
+    } else if (hasNegative && hasPositive) {
+      const y0 = Math.max(Math.abs(d3.min(data)), Math.abs(d3.max(data)));
+
+      domain = [-y0, y0];
```

Comment:
no, it is little more complicated than that. When we have positive and negative values, the chart needs to have the negative values going downward and the positive values going upward. So to make it work, I have two bar charts like this. The range for the bar chart positioned upside down needs to have negative values in the range.

<img width="283" alt="screen shot 2016-09-13 at 10 19 57 am" src="https://cloud.githubusercontent.com/assets/9920303/18482007/bb95d0ea-799b-11e6-8f11-b903010e4a27.png">